internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,250 @@
+            unique (scope_id),
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version
+);
+comment on table event_config is
+'event_config is a table where each row defines the event configuration for '
+'a scope.  Currently, the only supported scope is global';
+
+create trigger
+    update_version_column
+after update on event_config
+    for each row execute procedure update_version_column();
+
+create trigger
+    update_time_column
+before update on event_config
+    for each row execute procedure update_time_column();
+
+create trigger
+    default_create_time_column
+before insert on event_config
+    for each row execute procedure default_create_time();
+
+create table event_type_enabled (
+    config_id wt_public_id
+        constraint event_config_fkey
+            references event_config(public_id)
+            on delete cascade
+            on update cascade,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm (name)
+            on delete restrict
+            on update cascade,
+    primary key (config_id, event_type)
+);
+comment on table event_type_enabled is
+'event_type_enabled is a table where each row represents that eventing has '
+'been enabled for the specified event type in an event configuration';
+
+
+create table event_sink(
+    public_id wt_public_id primary key,
+    config_id wt_public_id 
+        constraint event_config_fkey
+            references event_config(public_id)
+            on delete cascade
+            on update cascade
+);
+comment on table event_sink is 
+'event_sink is a table where each row represents a configured event sink';
+
+create table event_file_sink(
+    public_id wt_public_id primary key,
+    sink_id wt_public_id not null
+        constraint sink_id_fkey
+            references event_sink(public_id)
+            on delete cascade
+            on update cascade,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    format_type text not null
+        constraint event_format_type_enm_fkey
+            references event_format_type_enm(name)
+            on delete restrict
+            on update cascade,
+    allow_filter wt_bexprfilter,
+    deny_filter wt_bexprfilter,
+    path text not null 
+        constraint path_not_empty
+            check (
+                length(trim(path)) > 0
+            ),
+    filename text not null
+        constraint filename_not_empty
+            check (
+                length(trim(filename)) > 0
+            ),
+        constraint path_filename_uq
+            unique(path, filename), -- ensure each sink is writing to a unique file
+    rotate_bytes int
+        constraint rotate_bytes_null_or_greater_than_zero
+            check(
+                rotate_bytes is null
```

Comment:
This is not needed.

> It should be noted that a check constraint is satisfied if the check expression evaluates to true or the null value.

-- [postgresql documentation](https://www.postgresql.org/docs/current/ddl-constraints.html)